odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -49,10 +50,46 @@ class OdooBaseChecker(BaseChecker):
             'gettext-repr',
             'Don\'t use %r to automatically insert quotes in translation strings. Quotes can be different depending on the language: they must be part of the translated string.',
         ),
+        'E8507': (
+            'Static string passed to %s without gettext call.',
+            'missing-gettext',
+            'Ensure that all static strings passed to certain constructs are wrapped in a gettext call.',
+        ),
     }
 
-    @only_required_for_messages('gettext-variable', 'gettext-placeholders', 'gettext-repr')
+    required_inside_functions = ['UserError', 'ValidationError', 'AccessError', 'AccessDenied', 'MissingError']
+
+    @only_required_for_messages('missing-gettext', 'gettext-variable', 'gettext-placeholders', 'gettext-repr')
     def visit_call(self, node):
+        file_path = self.linter.current_file
+        if (
+            "test" in file_path
+            and (module_path := find_module_root(file_path))
+            and (
+                # Check if the module's base name starts with "test_"
+                # This identifies modules dedicated to testing (e.g., test_mail)
+                os.path.basename(os.path.normpath(module_path)).startswith("test_")
+                # OR check if the file is in a "tests" subdirectory within the module
+                # This identifies test files located within the standard "tests" directory/package
+                or file_path[len(module_path) + 1 :].startswith("tests" + os.sep)
+            )
+        ):
```

Comment:
This can produce false positives like (fictive example): `/addons/my_gym/models/test_aptitude.py`, of course you could rename files like this.
What could be done instead is to call this check after the others (wrap it in a `def` and call before adding an exception).

Reply:
It's no big deal if I miss a few files, it just means they're not being checked. What I meant when I talked about false positives is that unless we realize we are missing a ton of files, better keep it simple 🙂

Reply:
It's acceptable to keep it simple here. If you change the code, please leave a comment indicating that there may be some missing cases.